src/internal/operations/test/restore_helper_test.go

```diff
@@ -0,0 +1,245 @@
+package test_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/alcionai/corso/src/internal/common/idname"
+	"github.com/alcionai/corso/src/internal/events"
+	evmock "github.com/alcionai/corso/src/internal/events/mock"
+	"github.com/alcionai/corso/src/internal/kopia"
+	"github.com/alcionai/corso/src/internal/m365"
+	"github.com/alcionai/corso/src/internal/m365/resource"
+	"github.com/alcionai/corso/src/internal/model"
+	"github.com/alcionai/corso/src/internal/operations"
+	"github.com/alcionai/corso/src/internal/streamstore"
+	"github.com/alcionai/corso/src/internal/tester/tconfig"
+	"github.com/alcionai/corso/src/pkg/account"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/control/repository"
+	"github.com/alcionai/corso/src/pkg/count"
+	"github.com/alcionai/corso/src/pkg/selectors"
+	"github.com/alcionai/corso/src/pkg/storage"
+	"github.com/alcionai/corso/src/pkg/store"
+)
+
+type restoreOpDependencies struct {
+	acct account.Account
+	ctrl *m365.Controller
+	kms  *kopia.ModelStore
+	kw   *kopia.Wrapper
+	sel  selectors.Selector
+	sss  streamstore.Streamer
+	st   storage.Storage
+	sw   *store.Wrapper
+
+	closer func()
+}
+
+func (rod *restoreOpDependencies) close(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+) {
+	fmt.Printf("\n-----\nREST CLOSING\n-----\n")
+
+	rod.closer()
+
+	if rod.kw != nil {
+		err := rod.kw.Close(ctx)
+		assert.NoErrorf(t, err, "kw close: %+v", clues.ToCore(err))
+	}
+
+	if rod.kms != nil {
+		err := rod.kw.Close(ctx)
+		assert.NoErrorf(t, err, "kms close: %+v", clues.ToCore(err))
+	}
+}
+
+// prepNewTestRestoreOp generates all clients required to run a restore operation,
+// returning both a resttore operation created with those clients, as well as
+// the clients themselves.
+func prepNewTestRestoreOp(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+	backupStore storage.Storage,
+	backupID model.StableID,
+	bus events.Eventer,
+	ctr *count.Bus,
+	sel selectors.Selector,
+	opts control.Options,
+	restoreCfg control.RestoreConfig,
+) (
+	operations.RestoreOperation,
+	*restoreOpDependencies,
+) {
+	var (
+		rod = &restoreOpDependencies{
+			acct: tconfig.NewM365Account(t),
+			st:   backupStore,
+		}
+		k = kopia.NewConn(rod.st)
+	)
+
+	err := k.Connect(ctx, repository.Options{})
+	require.NoError(t, err, clues.ToCore(err))
+
+	// kopiaRef comes with a count of 1 and Wrapper bumps it again
+	// we're so safe to close here.
+	rod.closer = func() {
+		err := k.Close(ctx)
+		assert.NoErrorf(t, err, "k close: %+v", clues.ToCore(err))
+	}
```

Comment:
no matter what it's safe to close k at the end of the function since it's not a part of the restore dependencies struct
```suggestion
	defer k.Close(ctx)
```